ResearchKit/Common/ORKHeightPicker.m

```diff
@@ -0,0 +1,272 @@
+    self = [super init];
+    if (self) {
+        NSAssert([answerFormat isKindOfClass:[ORKHeightAnswerFormat class]], @"answerFormat should be ORKHeightAnswerFormat");
+        
+        // Take into account locale changes so unit string can be updated
+        [[NSNotificationCenter defaultCenter] addObserver:self
+                                                 selector:@selector(currentLocaleDidChange:)
+                                                     name:NSCurrentLocaleDidChangeNotification object:nil];
+
+        self.answer = answer;
+        _answerFormat = answerFormat;
+        _pickerDelegate = delegate;
+    }
+    return self;
+}
+
+- (UIView *)pickerView {
+    if (_pickerView == nil) {
+        _pickerView = [[UIPickerView alloc] init];
+        _pickerView.dataSource = self;
+        _pickerView.delegate = self;
+        [self setAnswer:_answer];
+    }
+    return _pickerView;
+}
+
+- (void)setAnswer:(id)answer {
+    _answer = answer;
+    
+    if (ORKIsAnswerEmpty(answer)) {
+        answer = [self defaultAnswerValue];
+    }
+    
+    if (_answerFormat.useMetricSystem) {
+        NSUInteger index = [[self centimeterValues] indexOfObject:answer];
+        if (index == NSNotFound) {
+            [self setAnswer:[self defaultAnswerValue]];
+            return;
+        }
+        [_pickerView selectRow:index inComponent:0 animated:NO];
+    } else {
+        double feet, inches;
+        ORKInchesToFeetAndInches(((NSNumber *)answer).doubleValue, &feet, &inches);
+        NSUInteger feetIndex = [[self feetValues] indexOfObject:@((NSInteger)feet)];
+        NSUInteger inchesIndex = [[self inchesValues] indexOfObject:@((NSInteger)inches)];
+        if (feetIndex == NSNotFound || inchesIndex == NSNotFound) {
+            [self setAnswer:[self defaultAnswerValue]];
+            return;
+        }
+        [_pickerView selectRow:feetIndex inComponent:0 animated:NO];
+        [_pickerView selectRow:inchesIndex inComponent:1 animated:NO];
+    }
+}
+
+- (id)answer {
+    return _answer;
+}
+
+- (NSNumber *)defaultAnswerValue {
+    NSNumber *defaultAnswerValue = nil;
+    if (_answerFormat.useMetricSystem) {
+        defaultAnswerValue = @(162);
+    } else {
+        defaultAnswerValue = @(64); // 5ft 4in
+    }
+    return defaultAnswerValue;
+}
+
+- (NSNumber *)selectedAnswerValue {
+    NSNumber *answer = nil;
+    if (_answerFormat.useMetricSystem) {
+        NSInteger row = [_pickerView selectedRowInComponent:0];
+        answer = [self centimeterValues][row];
+    } else {
+        NSInteger feetRow = [_pickerView selectedRowInComponent:0];
+        NSInteger inchesRow = [_pickerView selectedRowInComponent:1];
+        NSNumber *feet = [self feetValues][feetRow];
+        NSNumber *inches = [self inchesValues][inchesRow];
+        answer = @( ORKFeetAndInchesToInches(feet.doubleValue, inches.doubleValue) );
```

Comment:
The thing is that the answer value is used to look for the selected row in the picker (`selectedRowInComponent:`), so I'm afraid that converting back and forth to centimeters might introduce floating point rounding errors.

Because of this, I figured out that it's better to use the locale format in the picker (which when are in inches it should not cause rounding errors), and just convert to *cm* when then answer result is calculated.